src/app/api/services/edit-page/edit-page.service.ts

```diff
@@ -175,4 +163,22 @@ export class EditPageService {
 
         return pageModeString[pageMode];
     }
+
+    private getPageWithExtraProperties(page: DotRenderedPage): DotRenderedPage {
+        const locked = !!page.lockedBy;
+        const lockedByAnotherUser = locked ? page.lockedBy !== this.loginService.auth.user.userId : false;
+
+        page = {
+            ...page,
+            locked,
+            lockedByAnotherUser
+        };
+
+        const mode: PageMode = page.lockedByAnotherUser ? PageMode.PREVIEW : this.getPageMode(page);
```

Comment:
Sending the page mode from the services so we don't have to calc this in the component or the toolbar